src/test/java/org/opensearch/security/system_indices/AbstractSystemIndicesTests.java

```diff
@@ -174,7 +174,6 @@ void validateSearchResponse(RestHelper.HttpResponse response, int expectedHits)
         assertThat(searchResponse.status(), is(RestStatus.OK));
         assertThat(searchResponse.getHits().getHits().length, is(expectedHits));
         assertThat(searchResponse.getFailedShards(), is(0));
-        assertThat(searchResponse.getSuccessfulShards(), is(5));
```

Comment:
I decided to remove this assertion because it wasn't providing much value. There are some nuances to this that make refactoring difficult. This could be refactored as `assertThat(searchResponse.getSuccessfulShards(), anyOf(is(5), is(1)));` but I don't think the shard count is core to the assertion here so its better to remove. What is important here is that the expectedHits matches the length of the returned hits.

The reason that the shard count can be either 5 or 1 is because it depends on if the cluster has a plugin that declares `SystemIndexPlugin.getSystemIndexDescriptor` or if a system index is declared solely in opensearch.yml with `plugins.security.system_indices.indices`. The tests in this suite do the latter which gives the indices being tested system index protection (which is the goal of the test), but from the perspective of the core they are not system indices because they were not registered with `SystemIndexPlugin.getSystemIndexDescriptors`. Since these tests are subclasses of SingleClusterTest/AbstractSecurityUnitTests its non-trivial to define a test plugin that declares these indices.

This logic is also covered in the SystemIntegratorsTests above [here](https://github.com/opensearch-project/security/pull/4867/files#diff-17ccaf0e8bfe3d0a06458dd9eff14212fe2c49f9eba4db6b23eddee502be74d1R292).

Reply:
The reason this test started failing is because a [PR was merged into core](https://github.com/opensearch-project/OpenSearch/pull/16418) to ignore index templates for system indices.

For testing, the security plugin defines a default template on all indices: https://github.com/opensearch-project/security/blob/main/src/test/java/org/opensearch/security/test/helper/cluster/ClusterHelper.java#L288-L304. I'm not sure why the security plugin does this, it would be better to refactor this in a more targeted way.